fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(classes=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    classes : dict, default=None
```

Comment:
I'm not super happy with the arg name `classes` since that's typically used for target classes (as in multi-class classification, for example). Groups is the term I'd prefer, but it's also a config dictionary for the groups, so perhaps `group_config`? I strongly suspect we'll expand this in the future to work for multiple sensitive features, so this arg will likely evolve.

This is also focused on binary classification right now (which is perfectly fine!), so perhaps we should include that in the name (similar to `sklearn`'s `make_classification`)?

@adrinjalali @hildeweerts @riedgar-ms @MiroDudik any thoughts on naming and API?

Reply:
I've replaced `classes` with `feature_config`. See https://github.com/fairlearn/fairlearn/pull/907#discussion_r671835815 .

I actually originally called it `make_classification`, rather than `make_synthetic_dataset`! I replaced it because I wasn't sure about overloading such a common function from scikit-learn. What about `make_sensitive_classification` or `make_gendered_classification`?

Reply:
It's certainly not an unreasonable thought. I'll wait for others to chime in (at least @adrinjalali may have thoughts), but `make_sensitive_classification` sounds like a name that'll work even with more sensitive features (even though that may not be part of the initial PR).